zipline/utils/calendars/trading_calendar.py

```diff
@@ -149,6 +149,16 @@ def open_offset(self):
     def close_offset(self):
         return 0
 
+    @lazyval
+    def minutes_per_session(self):
+        """
+        Returns
+        -------
+        df: A dataframe containing the number of minutes per session.
+        """
+        diff = self.schedule.market_close - self.schedule.market_open
+        return diff.astype('timedelta64[m]')
```

Comment:
Instead of the calendar exposing a dataframe to consumers, so now we're not in the managed land of sessions and labels anymore, would it make sense to expose a function that takes a list of sessions and does the slicing, i.e. move the call site code here?